File: prob60/tic_tac_toe_2.cs
// 実行時間 10ms
using static System.Console;

internal class Program
{
    static void Main()
    {
        for( int _ = 0; _ < 5; _++ ) {
            string s = ReadLine();
            var c = System.StringComparison.Ordinal;

            if( string.Compare( s, "OOOOO", c ) == 0 ) {
                WriteLine( 'O' );
                return;
            } else if( string.Compare( s, "XXXXX", c ) == 0 ) {
                WriteLine( 'X' );
                return;
            }
        }
        WriteLine( 'D' );
        return;
    }
}

File: prob60/tic_tac_toe_2.py
# 実行時間 10ms
import sys

for _ in [0] * 5:
    s = input()

    if s == "OOOOO":
        print("O")
        sys.exit()
    elif s == "XXXXX":
        print("X")
        sys.exit()
print("D")
